src/test/java/com/ncc/aif/TestUtils.java

```diff
@@ -0,0 +1,466 @@
+     * Returns a prefixed String URI of the specified name.
+     */
+    String getUri(String localName) {
+        return annotationNamespace + localName;
+    }
+
+    /**
+     * Returns a unique prefixed String URI for use with assertions.
+     */
+    String getAssertionUri() {
+        return getUri("assertion-" + assertionCount++);
+    }
+
+    /**
+     * Returns a unique prefixed String URI for use with entities.
+     */
+    String getEntityUri() {
+        return getUri("entity-" + entityCount++);
+    }
+
+    /**
+     * Returns a unique prefixed String URI for use with events.
+     */
+    String getEventUri() {
+        return getUri("event-" + eventCount++);
+    }
+
+    /**
+     * Returns a unique prefixed String URI for use with relations.
+     */
+    String getRelationUri() {
+        return getUri("relation-" + relationCount++);
+    }
+
+    /**
+     * Returns a unique prefixed String URI for use with hypotheses.
+     */
+    String getHypothesisUri() {
+        return getUri("hypothesis-" + hypothesisCount++);
+    }
+
+    /**
+     * Returns a unique prefixed String URI for use with clusters.
+     */
+    String getClusterUri() {
+        return getUri("cluster-" + clusterCount++);
+    }
+
+    /**
+     * Returns the test system created by {@link #startNewTest()}.
+     */
+    Resource getSystem() {
+        return system;
+    }
+
+    /**
+     * Returns a unique document name.
+     */
+    String getDocumentName() {
+        return "document-" + documentCount++;
+    }
+
+    /**
+     * Returns a type assertion justification.
+     */
+    Resource getTypeAssertionJustification() {
+        if (typeAssertionJustification == null) {
+            typeAssertionJustification = makeTextJustification(model, getDocumentName(),
+                    42, 143, system, 0.973);
+        }
+        return typeAssertionJustification;
+    }
+
+    /**
+     * Add the specified type to the specified resource.
+     *
+     * @param resource the object to which to add the type
+     * @param type     the type of the entity, event, or relation being asserted
+     * @return the created type assertion resource
+     */
+    Resource addType(Resource resource, Resource type) {
+        return addType(resource, type, 1.0);
+    }
+
+    /**
+     * Add the specified type to the specified resource with the specified confidence.
+     *
+     * @param resource   the object to which to add the type
+     * @param type       the type of the entity, event, or relation being asserted
+     * @param confidence the confidence with which to mark the specified type
+     * @return the created type assertion resource
+     */
+    Resource addType(Resource resource, Resource type, double confidence) {
+        return markType(model, getAssertionUri(), resource, type, system, confidence);
+    }
+
+    /**
+     * Makes and returns a valid AIF entity object of the specified type.
+     */
+    Resource makeEntity(Resource type) {
```

Comment:
Because these are supposed to be Examples as well as Tests, I would avoid making any method names in TestUtils have the same name as methods in AIFUtils.  It just makes this harder to read.  This is compounded by the fact that we use AIFUtils methods without the "AIFUtils." prefix.  So, I'd prefer method names like "makeValidBaseAIFEntity" and "makeValidNistEntity".  Likewise for makeEvent, etc.

Reply:
Hmm, I'm not sure I agree.   Initially I did it as you suggest (e.g., `makeValidEntity`, `makeValidEvent`), but I changed it to what you see here because:
- The reality is that the only true "example code" is the stuff Ryan wrote.  Pretty much everything we've written since was for the purpose of testing functionality (mostly restrictions) and generating AIF that "shows" how to do things.  If we really wanted example code for all of the restrictions, we probably wouldn't have written these utilities because it shortens code and you'd need to dig into the TestUtils classes to see what's really happening.
- I actually thought it was a feature not a bug to use the same names when I could...if you want to make an event you call...`makeEvent`.  It's not ambiguous (arguably) because you always invoke everything through your `utils` instance...there are no static methods.
- In fact, the only reason the Nist calls have the word Nist in them (e.g., `makeNistEvent`) was because originally there was no NistTestUtils subclass, and they had different return objects.  I really should call them `makeEvent`, etc.

Maybe a compromise of sorts is to call them all `makeValidThing()` in all three classes, without the word 'Nist' or "BaseAIF' in any of the method names.  I'd be curious to hear the others' thoughts on the matter, though.

Reply:
I renamed everything makeValidXXX().